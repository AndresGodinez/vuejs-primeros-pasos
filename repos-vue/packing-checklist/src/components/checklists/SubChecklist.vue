<template>
  <section class="subchecklist">
    <h1 class="subchecklist__title">{{title}}</h1>
    <ul>
      <li v-for="item in items" :key="item.id">
        <checklist-item :title="item.title" :done="item.done" :id="item.id" />
      </li>
    </ul>
  </section>
</template>

<script>
import ChecklistItem from '@/components/checklists/ChecklistItem'

export default {
  name: 'sub-checklist',
  props: ['title', 'items'],
  components: {
    ChecklistItem
  }
}
</script>

<style lang="sass" scoped>
.subchecklist__title
  margin-top: 1em
</style>
