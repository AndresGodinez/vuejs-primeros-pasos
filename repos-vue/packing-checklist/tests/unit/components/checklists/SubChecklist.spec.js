import { shallow } from '@vue/test-utils'
import SubChecklist from '@/components/checklists/SubChecklist.vue'

describe('SubChecklist', () => {
  it('has the right DOM structure', () => {
    const wrapper = shallow(SubChecklist, {
      propsData: { id: 0, title: 'waka', items: [] }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
