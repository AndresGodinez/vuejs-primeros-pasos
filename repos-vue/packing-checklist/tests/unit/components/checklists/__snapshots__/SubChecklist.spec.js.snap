// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubChecklist has the right DOM structure 1`] = `
<section
  class="subchecklist"
>
  <h1
    class="subchecklist__title"
  >
    waka
  </h1>
   
  <ul />
</section>
`;
